### DISCUSSION TITLE
Uninstalling Liboqs

### DISCUSSION THREAD

**Discussion by yamus-didine** [open-quantum-safe/liboqs#1536] [Q&A] [answered]
Hello.
I am looking for a way to "clean" and "uninstall" Liboqs. Is there such a thing?
To install it, I created a "build" directory, where I built Liboqs with custom options. Then, when I wanted to "re-install" it, I deleted the "build" directory and repeated the "build and install" process with OTHER options.
Is what I did right? did Liboqs get "uninstalled" and then "re-installed" with the new options?
Thanks in advance.

**Comment by baentsch** [comment]
If you never truly installed (`cmake --install`) `liboqs` then indeed, all binaries only reside in the "build" directory and deleting that directory brings you back to a pristine state.

**Reply by yamus-didine** [reply]
Edit:
After building in the first time, I did use `ninja` followed by `ninja install`.
Then, when I tried to uninstall, I deleted the build directory. I deleted `usr/include/oqs` directory and other directories and files in `usr/lib/` including `liboqs.so`, `liboqs.so.4` and `liboqs.0.9.0-dev` and then, I repeated the build-and-install process.

**Reply by baentsch** [reply]
> I did use ninja followed by ninja install.

Then indeed you installed `liboqs` in the standard system install locations on your machine. `liboqs` does not provide an "uninstall" `cmake` target -- and `cmake` in turn [does not provide this by default](https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#can-i-do-make-uninstall-with-cmake) -- but it might be a worthwhile addition (@SWilson4 @praveksharma @dstebila : What do you think?).

**Reply by yamus-didine** [reply]
Thanks for the reply.
So what would the final answer be ?
Do I proceed as I mentionned and delete those files and directories  in `usr/include` and `usr/lib/... ` ? 
I note that I also deleted  a directory named `cmake/liboqs` within `usr/lib/... `

**Comment by baentsch** [comment, marked as answer]
> So what would the final answer be ?

Remove all files listed in "install_manifest.txt" in the "build" directory: It contains the list of all files installed (as per the above, created during `cmake --install`).

**Reply by dstebila** [reply]
Seems reasonable to me, as long as CMake provides us a way to create that functionality.

**Reply by baentsch** [reply]
It does: See https://github.com/open-quantum-safe/liboqs/pull/1604
